### DISCUSSION TITLE
SP-5.39.10 and locales

### DISCUSSION THREAD

**Discussion by sisyphus** [StrawberryPerl/Perl-Dist-Strawberry#180] [General]
2 separate issues:
Firstly, on SP-5.39.10, I still get:
``` 
> perl -MPOSIX -wle "$loc = POSIX::setlocale( LC_ALL, 'English_UK.utf8' ); print $loc;"
Use of uninitialized value $loc in print at -e line 1.
```
Whereas, if 5.39.10 is built using the UCRT runtime (either gcc-13,  or VS-2022) I get:
```
>perl -MPOSIX -wle "$loc = POSIX::setlocale( LC_ALL, 'English_UK.utf8' ); print $loc;"
English_United Kingdom.utf8
```
Secondly, I'm not so sure that it's correct to describe the <b>"may hang or crash the interpreter"</b> issue with Asian locales as "fixed".
Sure, the warning is now issued only when the problematic locale is actually used (and only once), but the potential for a "hang" or "crash" still remains.

I have no  personal investment in these issues, and perhaps the points I've mentioned here are of little importance - which is the reason that I've chosen to raise  them here, rather than  further polluting other threads such as https://github.com/StrawberryPerl/Perl-Dist-Strawberry/issues/150.

Cheers,
Rob

**Comment by shawnlaffan** [comment]
Thanks Rob.  

Ideally we'll move to UCRT for perl 5.40.  I just need to find time to rebuild the external deps, and update a few.

**Comment by sisyphus** [comment]
> Ideally we'll move to UCRT for perl 5.40.

Excellent !! (I'll try to stop bangin' on about it, now.)
